### PR TITLE
HHH-11089: Invoke the physical strategy for foreign/index keys.

### PR DESCRIPTION
**Pull request by ptahchiev**
When creating foreign/index keys the physical strategy is not invoked
so the user has no way to change the resulting foreign/index key.
This prevents hibernate to be used with Oracle database where there's
restrictions on the length of foreign/index keys.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java [hunk 1]
```diff
@@ -1847,10 +1848,12 @@ public MetadataBuildingContext getBuildingContext() {
 								}
 							}
 					);
-
-					fk.setName( nameIdentifier.render( getDatabase().getJdbcEnvironment().getDialect() ) );
+				} else {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java]: Please follow code conventions and put the `else {` on a newline.

Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java]: As far as how to build the Identifier... ideally `ForeignKey` would carry the Identifier, like the other `org.hibernate.mapping` objects do.  The thing you'd have to do there is to make sure that `org.hibernate.mapping.Constraint` (the `ForeignKey` super) has access to the `org.hibernate.boot.spi.MetadataBuildingContext` the way other `org.hibernate.mapping` objects that need access to services do.

Short of that, you could rely on the `fk.getName()` call already being quoted using back-ticks or JPA's double-quotes.  The `Identifier.toIdentifier( fk.getName() )` call (no boolean) wold handle that case.  You could also hook in auto-quoting checks, if enabled.

File: hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java [hunk 2]
```diff
@@ -1847,10 +1848,12 @@ public MetadataBuildingContext getBuildingContext() {
...
+					nameIdentifier = Identifier.toIdentifier( fk.getName(), true );
 				}
 
+				fk.setName( getMetadataBuildingOptions().getPhysicalNamingStrategy().toPhysicalSequenceName(nameIdentifier, getDatabase().getJdbcEnvironment()).render( getDatabase().getJdbcEnvironment().getDialect() ) );
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java]: Totally unexpected to route this through `#toPhysicalSequenceName`.  A FK is not a sequence ;)

File: hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java (hunk 1 above)
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java]: To clarify... "Short of that, you could rely on the fk.getName() call already being quoted using back-ticks or JPA's double-quotes *in cases where the user wants a quoted name*"

File: hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java (hunk 2 above)
Comment (ptahchiev) [inline hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java]: Hi Steve,

what method should I route it through then? Maybe `ImplicitNamingStrategy.determineForeignKeyName`?

**Comment by vladmihalcea** [comment]
Closing PR since the issue was fixed already.

**Pull request closed without merging**